File: examples/python/guides/integrations/scraper_openai.py
# Third-party integration example - requires: pip install openai
# See: /guides/web-scraping

from openai import OpenAI

client = OpenAI()


# > OpenAI web search usage
def search_and_extract(query: str) -> dict:
    response = client.responses.create(
        model="gpt-4o-mini",
        tools=[{"type": "web_search"}],
        input=query,
    )
    return {"query": query, "content": response.output_text}

File: examples/typescript/guides/integrations/scraper-openai.ts
// Third-party integration - requires: pnpm add openai
// See: /guides/web-scraping

import OpenAI from 'openai';

const openai = new OpenAI();

// > OpenAI web search usage
export async function searchAndExtract(query: string) {
  const response = await openai.responses.create({
    model: 'gpt-4o-mini',
    tools: [{ type: 'web_search' }],
    input: query,
  });
  return { query, content: response.output_text };
}
